File: Sample/Sample/HomeViewController.m
/* Copyright Airship and Contributors */

@import AirshipCore;

#import "HomeViewController.h"

@interface HomeViewController ()

@end

@implementation HomeViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(refreshView)
                                                 name:UAChannelUpdatedEvent
                                               object:nil];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];

    [self refreshView];
}

- (IBAction)buttonTapped:(id)sender {
    if (sender == self.enablePushButton) {
        [UAirship push].userPushNotificationsEnabled = YES;
        [[UAirship shared].privacyManager enableFeatures:UAFeaturesPush];
    }

    if (sender == self.channelIDButton && [UAirship channel].identifier) {
        [UIPasteboard generalPasteboard].string = [UAirship channel].identifier;

        UIAlertController* alert = [UIAlertController alertControllerWithTitle:nil
                                                                       message:NSLocalizedStringFromTable(@"UA_Copied_To_Clipboard", @"UAPushUI", @"Copied to clipboard string")
                                                                preferredStyle:UIAlertControllerStyleAlert];

        UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"UA_OK", @"UAPushUI", @"OK button string")
                                                           style:UIAlertActionStyleDefault
                                                         handler:^(UIAlertAction * _Nonnull action) {
                                                             [self dismissViewControllerAnimated:YES completion:nil];
                                                         }];

        [alert addAction:okAction];

        [self presentViewController:alert animated:YES completion:nil];
    }

    [self refreshView];
}

- (void)refreshView {
    if ([self checkNotificationsEnabled]) {
        [self.channelIDButton setTitle:[UAirship channel].identifier forState:UIControlStateNormal];
        self.channelIDButton.hidden = NO;
        self.enablePushButton.hidden = YES;
        return;
    }
    self.channelIDButton.hidden = YES;
    self.enablePushButton.hidden = NO;
}

- (BOOL)checkNotificationsEnabled {
    return [UAirship push].userPushNotificationsEnabled && [[UAirship shared].privacyManager isEnabled:UAFeaturesPush];
}
@end

File: Sample/Sample/HomeViewController.swift
/* Copyright Airship and Contributors */

import UIKit
import AirshipCore
import AirshipPreferenceCenter

class HomeViewController: UIViewController {

    @IBOutlet private weak var enablePushButton: UIButton!
    @IBOutlet private weak var channelIDButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()

        NotificationCenter.default.addObserver(
            self,
            selector: #selector(HomeViewController.refreshView),
            name: Channel.channelCreatedEvent,
            object: nil);
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        refreshView()
    }

    @objc func refreshView () {
        if (checkNotificationsEnabled()) {
            channelIDButton.setTitle(Airship.channel.identifier, for: [])
            channelIDButton.isHidden = false
            enablePushButton.isHidden = true
            return
        }

        channelIDButton.isHidden = true
        enablePushButton.isHidden = false
    }

    @IBAction func buttonTapped(_ sender: UIButton) {
        if (sender == enablePushButton) {
            Airship.push.userPushNotificationsEnabled = true
            Airship.shared.privacyManager.enableFeatures(.push)
        }

        //The channel ID will need to wait for push registration to return the channel ID
        if (sender == channelIDButton) {
            if ((Airship.channel.identifier) != nil) {
                UIPasteboard.general.string = Airship.channel.identifier

                let message = NSLocalizedString("UA_Copied_To_Clipboard", tableName: "UAPushUI", comment: "Copied to clipboard string")

                let alert = UIAlertController(title: nil, message: message, preferredStyle: UIAlertController.Style.alert)
                let buttonTitle = NSLocalizedString("UA_OK", tableName: "UAPushUI", comment: "OK button string")

                let okAction = UIAlertAction(title: buttonTitle, style: UIAlertAction.Style.default, handler: { (action: UIAlertAction) in
                    self.dismiss(animated: true, completion: nil)
                })

                alert.addAction(okAction)

                self.present(alert, animated: true, completion: nil)
            }
        }

        refreshView()
    }

    func checkNotificationsEnabled() -> Bool {
        if (!Airship.push.userPushNotificationsEnabled) {
            return false
        }

        return Airship.shared.privacyManager.isEnabled(.push)
    }
}

